src/track/timesignature.h

```diff
@@ -0,0 +1,49 @@
+// signature.h
+// Created on: 06/11/2019 by Javier Vilarroig
+
+#pragma once
+
+namespace mixxx {
+
+// Musical Time signature
+// Right now is to be used for bar detection only, so only the beats per bar is useful
+class TimeSignature final {
+  private:
+    int m_beatsPerBar;
+    int m_noteValue;
+
+  public:
+    inline TimeSignature(int beatsPerBar, int noteValue)
+        : m_beatsPerBar(beatsPerBar),
+          m_noteValue(noteValue) {
+    }
+
+    virtual ~TimeSignature() {}
+
+    inline void setBeats(int beatsPerBar) {
+        m_beatsPerBar = beatsPerBar;
+    }
+
+    inline void setNoteValue(int noteValue) {
+        m_noteValue = noteValue;
+    }
+
+    inline int getBeats() const {
+        return m_beatsPerBar;
+    }
+
+    inline int getNoteValue() const {
+        return m_noteValue;
+    }
+
+    inline bool operator==(const TimeSignature& signature) const {
```

Comment:
Please define all symmetric operators as free functions. Define both == and !=

Reply:
done